planner/core/exhaust_physical_plans.go

```diff
@@ -438,7 +525,13 @@ func (p *LogicalJoin) getIndexJoinByOuterIdx(prop *property.PhysicalProperty, ou
 			innerPlan := p.constructInnerTableScan(ds, pkCol, outerJoinKeys, us)
 			// Since the primary key means one value corresponding to exact one row, this will always be a no worse one
 			// comparing to other index.
-			return p.constructIndexJoin(prop, outerIdx, innerPlan, nil, keyOff2IdxOff, nil, nil)
+			indexJoins := p.constructIndexJoin(prop, outerIdx, innerPlan, nil, keyOff2IdxOff, nil, nil)
```

Comment:
do we still need to consider `PhysicalIndexJoin`? Seems `IndexMergeJoin` is no worse than `IndexJoin`.

Reply:
Actually not, the `IndexMergeJoin` must make its `innerPlan` keep order.